WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/Subscription/SubscriptionExpiryViewModelTests.swift

```diff
@@ -15,6 +15,16 @@ final class SubscriptionExpiryViewModelTests: XCTestCase {
         XCTAssertEqual(viewModel.selectedLength.stringValue, "0")
     }
 
+    func test_selectedLength_and_lengthOptions_are_correct_if_periodInterval_is_0() {
+        // Given
+        let subscription = ProductSubscription.fake().copy(periodInterval: "0")
+        let viewModel = SubscriptionExpiryViewModel(subscription: subscription) { _, _ in }
+
+        // Then
+        XCTAssertEqual(viewModel.selectedLength.value, 0)
+        XCTAssertEqual(viewModel.lengthOptions.count, 1)
+    }
```

Comment:
Question: Should we add an additional test to check for negative numbers as well, just for safety? Currently will work nicely because of the `periodInterval > 0` check, but if we happen to change the implementation for `periodInterval != 0` for some reason, this test would still pass but a negative number into the stride would also cause a crash.

Reply:
Thanks for the suggestion! We don't expect period interval to be less than 0, and it's unlikely that a plugin would make this happen, so I figure it's not necessary to cover that case.